app/models/spree/payment/processing.rb

```diff
@@ -6,14 +6,22 @@ module Processing
       def process!
         return unless validate!
 
-        purchase!
+        if response_code
```

Comment:
I think wrapping this in a more abstract query method such as `authorized?` would help a great deal in showing what's the actual "algorithm". As is, when I read this, I need to find out what does having a response_code at this point means, and I feel a bit lost. Didn't we discuss and implement this before?

Reply:
That's a good point. It sounds familiar because we did it for cvv_response_message:

```
def authorization_action_required?
   cvv_response_message.present?
end
```

I'll add it here too

Reply:
I'm not sure I'm following. We always call `authorize!` first and we call capture if required user authorization?

Reply:
We call authorize first, but the user authorization step is often not necessary. In either case (whether the user authorization step is required or not), we've already created a Payment Intent in Stripe by calling authorize, so this PR calls capture on that Payment Intent instead of creating a new one.

Reply:
I fail to see when would `#purchase!` create a new payment intent :thinking: In `#purchase` we `#fetch_payment_intent` first but both call `provider.capture` at the end. What am I missing?

Reply:
Hm yeah I think you're right actually. `#purchase` already guards against this, so it's just `#charge_offline` that we need to modify.

Reply:
again, having a regression test that we can rely on would make it easier for you to move code around while not breaking the behavior.